From bdc0419810857a5b2a45571e4ff0a3569e5e1004 Mon Sep 17 00:00:00 2001
Message-Id: <bdc0419810857a5b2a45571e4ff0a3569e5e1004.1421808898.git.chang-joon.lee@intel.com>
In-Reply-To: <dcec99b5d6fee3bcdab208c48cae68cdc752e9d4.1421808898.git.chang-joon.lee@intel.com>
References: <dcec99b5d6fee3bcdab208c48cae68cdc752e9d4.1421808898.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 11 Dec 2014 11:49:29 +0530
Subject: [PATCH 4/5] FROM_UPSTREAM [VPG]: drm/i915: Forcewake Register Range
 changes for CHV
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

According to updated BSpec, Render/Common Wells register range has
changed. Updating the same to match the spec and avoid extra forcewake
for none forcewake range.

Issue GMINL-4685
Change-Id: I189805bfab945396a33c8faaa10d4c940e059ee1
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 5cb7179b428cf00dd96e74ed4f37efb19ba05e14
Upstream-as-of-tag: drm-intel-nightly
Reviewed-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_uncore.c |   11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_uncore.c b/drivers/gpu/drm/i915/intel_uncore.c
index 93c2d46..3d589c6 100644
--- a/drivers/gpu/drm/i915/intel_uncore.c
+++ b/drivers/gpu/drm/i915/intel_uncore.c
@@ -526,9 +526,9 @@ void assert_force_wake_inactive(struct drm_i915_private *dev_priv)
 
 #define FORCEWAKE_CHV_RENDER_RANGE_OFFSET(reg) \
 	(REG_RANGE((reg), 0x2000, 0x4000) || \
-	 REG_RANGE((reg), 0x5000, 0x8000) || \
+	 REG_RANGE((reg), 0x5200, 0x8000) || \
 	 REG_RANGE((reg), 0x8300, 0x8500) || \
-	 REG_RANGE((reg), 0xB000, 0xC000) || \
+	 REG_RANGE((reg), 0xB000, 0xB480) || \
 	 REG_RANGE((reg), 0xE000, 0xE800))
 
 #define FORCEWAKE_CHV_MEDIA_RANGE_OFFSET(reg) \
@@ -537,17 +537,14 @@ void assert_force_wake_inactive(struct drm_i915_private *dev_priv)
 	 REG_RANGE((reg), 0x12000, 0x14000) || \
 	 REG_RANGE((reg), 0x1A000, 0x1C000) || \
 	 REG_RANGE((reg), 0x1E800, 0x1EA00) || \
-	 REG_RANGE((reg), 0x30000, 0x40000))
+	 REG_RANGE((reg), 0x30000, 0x38000))
 
 #define FORCEWAKE_CHV_COMMON_RANGE_OFFSET(reg) \
 	(REG_RANGE((reg), 0x4000, 0x5000) || \
 	 REG_RANGE((reg), 0x8000, 0x8300) || \
 	 REG_RANGE((reg), 0x8500, 0x8600) || \
 	 REG_RANGE((reg), 0x9000, 0xB000) || \
-	 REG_RANGE((reg), 0xC000, 0xC800) || \
-	 REG_RANGE((reg), 0xF000, 0x10000) || \
-	 REG_RANGE((reg), 0x14000, 0x14400) || \
-	 REG_RANGE((reg), 0x22000, 0x24000))
+	 REG_RANGE((reg), 0xF000, 0x10000))
 
 static void
 ilk_dummy_write(struct drm_i915_private *dev_priv)
-- 
1.7.9.5

